plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionFields.java

```diff
@@ -29,10 +29,13 @@
 
 public final class PartitionFields
 {
-    private static final String NAME = "[a-z_][a-z0-9_]*";
+    private static final String IDENTIFIER = "[[a-z]_][[a-z0-9]_]*";
+    private static final String QUOTED_IDENTIFIER = "(?:\"[^\"]*\")+";
```

Comment:
a column name can contain a quotation mark itself (`"`).
in SQL, it is denoted by repeating the character `"a column with "" quotation mark"`

Reply:
It was indeed only partially implemented. I have adjusted the regex and added additional test cases. Build failure doesn't seem related `java.util.concurrent.TimeoutException: Idle timeout 5000 ms`. Build was green locally.